src/server/src/server.cpp

```diff
@@ -0,0 +1,59 @@
+#include <ros/ros.h>
+#include <client/time.h>
+#include <client/show.h>
+
+#include <vector>
+#include <iostream>
+
+client::show srv;
+std::vector<ros::Subscriber> server_subscribers;
+ros::ServiceServer service_server;
+
+void time_callback(const client::time::ConstPtr& time)
+{
+    ROS_INFO("name:%s   time:%s ", time->name.c_str(), time->sec.c_str());
+}
+
+// service回调函数，输入参数request，输出参数response
+bool show_callback(client::show::Request &request, client::show::Response &response)
+{
+    if(request.request == 1) 
+    {
+        ROS_INFO("A client has logged in!");
+        response.response = 10;
+        ros::NodeHandle n;
+        server_subscribers.push_back(n.subscribe(request.node_name, 10, time_callback));
+    }
+    else
+    {
+        ROS_INFO("A client has logged out!");
+
+        for(auto &each_subscriber : server_subscribers)
+        {
+            std::string assist_sig = "/";
+
+            if(assist_sig + request.node_name.c_str() == each_subscriber.getTopic())
```

Comment:
**[cpplint]** <sub>reported by [reviewdog](https://github.com/reviewdog/reviewdog) :dog:</sub><br> Lines should be <= 80 characters long  [whitespace/line_length] [2]